lib/rubocop/cop/style/yoda_expression.rb

```diff
@@ -0,0 +1,45 @@
+# frozen_string_literal: true
+
+module RuboCop
+  module Cop
+    module Style
+      # Forbids Yoda expressions,
+      # i.e. binary operations where the order of expression is reversed, eg. `1 + x`
+      #
+      # @safety
+      #   This cop is unsafe because binary operators can be defined
+      #   differently on different classes, and are not guaranteed to
+      #   have the same result if reversed.
+      #
+      # @example
+      #   # bad
+      #   1 + x
+      #   1 & z
+      #
+      #   # good
+      #   60 * 24
+      #   x + 1
+      #   10 * y
+      #   y * 10
+      #   z & 1
+      #
+      class YodaExpression < Base
+        extend AutoCorrector
+
+        MSG = 'Reverse the order of the operands.'
```

Comment:
Perhaps the message should make it clear the intention is for the non-literal operand to go first?